Add some basic xnnpack recipes

tarun292 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #10035
* #10034

Differential Revision: [D72085170](https://our.internmc.facebook.com/intern/diff/D72085170/)

tarun292 committed:
Recipe and Input class definitions with e2e export



Differential Revision: [D71946730](https://our.internmc.facebook.com/intern/diff/D71946730/)

[ghstack-poisoned]

tarun292 committed:
Add some basic xnnpack recipes



Differential Revision: [D72085170](https://our.internmc.facebook.com/intern/diff/D72085170/)

[ghstack-poisoned]

kimishpatel commented on backends/xnnpack/recipes/recipes.py:
```diff
@@ -0,0 +1,51 @@
+
+    )
+
+def get_dynamic_quant_recipe() -> ExportRecipe:
```
organizationally maybe quant recipes can be in a separate folder?

digantdesai commented on backends/xnnpack/recipes/recipes.py:
```diff
@@ -0,0 +1,51 @@
+)
+from executorch.exir import ExportRecipe
+
+def get_generic_fp32_cpu_recipe() -> ExportRecipe:
```
if namespaced to XNNPACK then cpu may not be needed?
```suggestion
def get_fp32_recipe() -> ExportRecipe:
```

digantdesai commented on backends/xnnpack/recipes/recipes.py:
```diff
@@ -0,0 +1,51 @@
+    )
+
+RECIPE_MAP: dict[str, Callable[[], ExportRecipe]] = {
+    "FP32_CPU_ACCELERATED_RECIPE": get_generic_fp32_cpu_recipe,
+    "DYNAMIC_QUANT_CPU_ACCELERATED_RECIPE": get_dynamic_quant_recipe,
```
nit
```suggestion
    "FP32_RECIPE": get_fp32_recipe,
    "DYNAMIC_QUANT_RECIPE": get_dynamic_quant_recipe,
```

digantdesai commented on backends/xnnpack/recipes/recipes.py:
```diff
@@ -0,0 +1,51 @@
+       name = "fp32_recipe",
+       quantizer = None,
+       partitioners=[XnnpackPartitioner()],
+
```
```suggestion
```

digantdesai commented on backends/xnnpack/recipes/recipes.py:
```diff
@@ -0,0 +1,51 @@
+from executorch.exir import ExportRecipe
+
+def get_generic_fp32_cpu_recipe() -> ExportRecipe:
+    quantizer = XNNPACKQuantizer()
+    operator_config = get_symmetric_quantization_config(is_per_channel=False)
+    quantizer.set_global(operator_config)
```
```suggestion

```

digantdesai commented on backends/xnnpack/recipes/recipes.py:
```diff
@@ -0,0 +1,51 @@
+        is_per_channel=True, is_dynamic=True
+    )
+    quantizer.set_global(operator_config)
+    DuplicateDynamicQuantChainPass
```
Let's remove it if not needed?

digantdesai commented on backends/xnnpack/recipes/recipes.py:
```diff
@@ -0,0 +1,51 @@
+    operator_config = get_symmetric_quantization_config(is_per_channel=False)
+    quantizer.set_global(operator_config)
+    return ExportRecipe(
+       name = "fp32_recipe",
```
nit
 ```suggestion
       name = "fp32",
```

digantdesai commented on backends/xnnpack/test/recipes/test_xnnpack_recipes.py:
```diff
@@ -0,0 +1,37 @@
+from torch.testing._internal.common_quantization import TestHelperModules
+
+class TestXnnpackRecipes(unittest.TestCase):
+    def setUp(self) -> None:
+        super().setUp()
+
+    def tearDown(self) -> None:
+        super().tearDown()
```
do we need these?

digantdesai commented on backends/xnnpack/test/recipes/test_xnnpack_recipes.py:
```diff
@@ -0,0 +1,37 @@
+            example_inputs=example_inputs, 
+            export_recipe=get_xnnpack_recipe("FP32_CPU_ACCELERATED_RECIPE")
+        )
+        export_session.export()
```
will this raise? should we catch it?

digantdesai reviewed (approved):
Looks good. Stamping, left some comments.

tarun292 committed:
Update base for Update on "Add some basic xnnpack recipes"


Differential Revision: [D72085170](https://our.internmc.facebook.com/intern/diff/D72085170/)

[ghstack-poisoned]

tarun292 committed:
Update on "Add some basic xnnpack recipes"


Differential Revision: [D72085170](https://our.internmc.facebook.com/intern/diff/D72085170/)

[ghstack-poisoned]

tarun292 committed:
Update base for Update on "Add some basic xnnpack recipes"


Differential Revision: [D72085170](https://our.internmc.facebook.com/intern/diff/D72085170/)

[ghstack-poisoned]

tarun292 committed:
Update on "Add some basic xnnpack recipes"


Differential Revision: [D72085170](https://our.internmc.facebook.com/intern/diff/D72085170/)

[ghstack-poisoned]

tarun292 committed:
Update base for Update on "Add some basic xnnpack recipes"


Differential Revision: [D72085170](https://our.internmc.facebook.com/intern/diff/D72085170/)

[ghstack-poisoned]

tarun292 committed:
Update on "Add some basic xnnpack recipes"


Differential Revision: [D72085170](https://our.internmc.facebook.com/intern/diff/D72085170/)

[ghstack-poisoned]

tarun292 committed:
Update base for Update on "Add some basic xnnpack recipes"


Differential Revision: [D72085170](https://our.internmc.facebook.com/intern/diff/D72085170/)

[ghstack-poisoned]

tarun292 committed:
Update on "Add some basic xnnpack recipes"


Differential Revision: [D72085170](https://our.internmc.facebook.com/intern/diff/D72085170/)

[ghstack-poisoned]

tarun292 committed:
Merge branch 'main' into gh/tarun292/5/head
